### ISSUE TITLE
Incorrect SETTINGS part silently ignored if it is a part of subquery.

### ISSUE THREAD

**Issue by qoega** [ClickHouse/ClickHouse#67617]
This is expected:
```
:) select 1 from ( select 2 ) settings join_using_nulls=1

SELECT 1
FROM
(
    SELECT 2
)
SETTINGS join_using_nulls = 1

Query id: 41fbcd14-6406-4a40-be90-360a50759d93


Elapsed: 0.001 sec.

Received exception:
Code: 115. DB::Exception: Unknown setting 'join_using_nulls': Maybe you meant ['join_use_nulls']. (UNKNOWN_SETTING)
```

This is not:
```
:) select 1 from ( select 2 settings join_using_nulls=1)

SELECT 1
FROM
(
    SELECT 2
    SETTINGS join_using_nulls = 1
)

Query id: a8fcf3f0-684a-461f-b599-322222b09f92

   ┌─1─┐
1. │ 1 │
   └───┘

1 row in set. Elapsed: 0.001 sec.
```

**Comment by qoega** [comment]
@groeneai can you reproduce it ?

**Comment by groeneai** [comment]
Yes, on master `26.9.1.1` (`88dd163466bc`), in `clickhouse-client` against a server built from that commit. Your exact pair behaves exactly as you reported.

It is a bit wider than the report:

* Not only unknown names. `select 1 from (select 2 settings max_block_size='abc')` also returns `1`; the same clause at statement level gives `Code: 27`.
* Same inside a CTE and inside an `IN` subquery.
* It persists on disk. `create view v as select 1 from (select 2 settings join_using_nulls=1)` succeeds, and `show create view v` gives the clause back verbatim, so a setting name that does not exist ends up stored in the view definition.

The clause is still honoured when it is valid, so it is only the validation that is skipped: `select count() from (select number from numbers(1000) settings max_rows_to_read=10)` gives `Code: 158`.

Mechanism on master today (line numbers at `88dd163466bc`). A nested clause is clamped, then applied, at `src/Analyzer/QueryTreeBuilder.cpp:316-318`. `SettingsConstraints::checkOrClamp` erases every change whose check fails (`src/Access/SettingsConstraints.cpp:315`), and in the non-throwing reaction that check returns `false` both for a name `isSettingNameAllowed` rejects (`:432`) and for a value that does not cast (`:436`). The change is gone before `applySettingsChanges`, which is the call that raises `Code: 115`. At statement level the same check runs in the throwing reaction, which is also where the `Maybe you meant` hint comes from.

That erase is deliberate for constraint violations, so this is not simply a missing throw. It is visible on its own: with `set readonly=1`, an outer `settings max_block_size=100` gives `Code: 164`, the same clause nested is silent, and the `max_rows_to_read=10` query above then returns `1000` instead of failing. So a fix has to separate a typo from a constraint violation.

Why I think it is worth fixing: it rots tests. A test that pins a setting inside a subquery keeps passing when the name is misspelled or is later renamed, so the pin quietly stops existing. I had one of those live for four months before noticing.

I have not proposed a change here, since rejecting these in nested position turns currently green queries and stored views into errors. Happy to take it if you want it.
